### PR TITLE
feat: temp caching transactions in TransactionStore

### PR DESCRIPTION
**Pull request by iGroza**
Fixes #

## Proposed Changes

-  temp caching transactions in TransactionStore

## Checklist

- [x] I have read the rules of [**contribution**](https://github.com/haqq-network/haqq-wallet/blob/main/docs/contribution.md#important-steps).

### CODE REVIEW

File: src/models/transaction.ts [hunk 1]
```diff
@@ -177,18 +184,29 @@ class TransactionStore {
           blockNumber === 'latest' &&
           this._lastSyncedAccountsHash !== accountHash
         ) {
-          this._transactions = [];
+          const cahced = this._cache.get(`${accountHash}:${blockNumber}`);
```
Comment (ragozin-nikita) [inline src/models/transaction.ts:187]:
```suggestion
          const cached = this._cache.get(`${accountHash}:${blockNumber}`);
```

File: src/models/transaction.ts [hunk 2]
```diff
@@ -37,12 +39,17 @@ export type Transaction = IndexerTransaction & {
...
+
 class TransactionStore {
   realmSchemaName = TransactionRealmObject.schema.name;
   private _transactions: Array<Transaction> = [];
-  private _lastSyncedAccountsHash = '';
-  private _lastSyncedBlockNumber = 'latest';
+  private _lastSyncedAccountsHash: AccountsHash = '';
+  private _lastSyncedBlockNumber: BlockNumber = 'latest';
   private _isLoading = false;
+  private _cache = new Map<CacheKey, Transaction[]>();
```
Comment (ragozin-nikita) [inline src/models/transaction.ts:52]: Is it working ok with `makeAutoObservable`?

File: src/models/transaction.ts [hunk 3]
```diff
@@ -177,18 +184,29 @@ class TransactionStore {
...
+      const result = await Indexer.instance.getTransactions(
         accounts,
         blockNumber,
         app.providerId,
       );
+      await when(() => !Token.isLoading, {});
+      const parsed = result
+        .map(tx => parseTransaction(tx, accounts))
+        .filter(tx => !!tx.parsed);
+      this._cache.set(`${this._lastSyncedAccountsHash}:${blockNumber}`, parsed);
```
Comment (ragozin-nikita) [inline src/models/transaction.ts:208]: I think it should be better to store key generation logic somewhere
```suggestion
      this._cache.set(this.generateKey(accounts, blockNumber), parsed);
```

File: src/models/transaction.ts (hunk 1 above)
Comment (ragozin-nikita) [inline src/models/transaction.ts:187]:
```suggestion
          const cahced = this._cache.get(this.generateKey(accounts, blockNumber));
```

**Review by ragozin-nikita — CHANGES_REQUESTED** [review]

File: src/models/transaction.ts (hunk 2 above)
Comment (iGroza) [reply src/models/transaction.ts:52]: yep

**Review by ragozin-nikita — APPROVED** [review]

**Pull request merged**